ethdk/src/EOA/EoaAccount.ts

```diff
@@ -0,0 +1,93 @@
+import { ethers, Wallet } from 'ethers'
+import { type Deferrable } from 'ethers/lib/utils'
+import type Account from '../interfaces/Account'
+import { type EoaNetwork, type Network } from '../interfaces/Network'
+import type Transaction from '../interfaces/Transaction'
+import { getNetwork } from './EoaNetworks'
+import EoaTransaction from './EoaTransaction'
+
+export default class EoaAccount implements Account {
+  accountType: string = 'eoa'
+
+  address: string
+  private readonly privateKey: string
+  private readonly recoveryPhrase: string
+  private readonly networkConfig: EoaNetwork
+  private readonly provider: ethers.providers.JsonRpcProvider
+  private readonly signer: Wallet
+
+  private constructor({
+    address,
+    privateKey,
+    recoveryPhrase,
+    network,
+    provider,
+    signer,
+  }: {
+    address: string
+    privateKey: string
+    recoveryPhrase: string
+    network: EoaNetwork
+    provider: ethers.providers.JsonRpcProvider
+    signer: Wallet
+  }) {
+    this.address = address
+    this.privateKey = privateKey
+    this.recoveryPhrase = recoveryPhrase
+    this.networkConfig = network
+    this.provider = provider
+    this.signer = signer
+  }
+
+  static async createAccount({
+    privateKey,
+    network,
+  }: {
+    privateKey?: string
+    network?: Network
+  } = {}): Promise<EoaAccount> {
```

Comment:
Added empty object default value here so the account can be created without passing in any parameters:
```ts
const account = await EoaAccount.createAccount()
```